src/nile/core/types/transactions.py

```diff
@@ -67,8 +67,12 @@ def __post_init__(self):
         # Validate the transaction object
         self._validate()
 
-    async def execute(self, signer, max_fee=None, watch_mode=None, **kwargs):
+    async def execute(self, signer, watch_mode=None, **kwargs):
         """Execute the transaction."""
+        if self.max_fee == 0:
+            max_fee = await self.estimate_fee(signer)
+            self.update_fee(max_fee)
```

Comment:
I also think that we should let the user set the `max_fee` as 0, that's why I think "auto" would be a better option for auto fee estimation. Then we know the user won't receive a different transaction from the one he was expecting (it could be one with max_fee equal 0, even if it doesn't make much sense to us, it could make sense).

Reply:
Ah forgive me, I lost sight of the tx.hash changing. The issue I was running into with that approach is `estimate_fee` is async, so I think executing this in during construction (with the sync `__post_init__`) won't work without getting into hacky territory. Because of the async issue, I don't think we can pass `auto` or `-1` to the `Transaction` during construction either.

Two options that I see are:
1. Have the users use `tx.update_fee` which IMO isn't automating the fee but as you mentioned, maybe it's enough
2. Automate the fee in the account methods which keeps the tx.hash consistent for the user

Reply:
Number 2 sounds great. Returning the tx_wrapper with the estimated fee if max_fee is left to None in account methods.

Reply:
On it!